modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2PaginationTestCase.java

```diff
@@ -0,0 +1,307 @@
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.SERVER_URL;
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.TYPE_PARAM;
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.USER_NAME_ATTRIBUTE;
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.VALUE_PARAM;
+
+public class SCIM2PaginationTestCase extends ISIntegrationTest {
+
+    public static final String TOTAL_RESULTS_ATTRIBUTE = "totalResults";
+    public static final String RESOURCES_ATTRIBUTE = "Resources";
+    public static final String START_INDEX_ATTRIBUTE = "startIndex";
+    public static final String ITEMS_PER_PAGE_ATTRIBUTE = "itemsPerPage";
+    public static final String START_INDEX = "2";
+    public static final String COUNT = "4";
+    private final String adminUsername;
+    private final String adminPassword;
+    private final String tenant;
+    List<String> userIds = new ArrayList<>();
+    UserManagementClient userMgtClient;
+    AutomationContext automationContext;
+    String backendUrl = null;
+    String sessionCookie = null;
+    private CloseableHttpClient client;
+
+    @Factory(dataProvider = "scim2UserConfigProvider")
+    public SCIM2PaginationTestCase(TestUserMode userMode) throws Exception {
+
+        AutomationContext context = new AutomationContext("IDENTITY", userMode);
+        this.adminUsername = context.getContextTenant().getTenantAdmin().getUserName();
+        this.adminPassword = context.getContextTenant().getTenantAdmin().getPassword();
+        this.tenant = context.getContextTenant().getDomain();
+
+        automationContext = new AutomationContext("IDENTITY", userMode);
+        backendUrl = automationContext.getContextUrls().getBackEndUrl();
+        sessionCookie = new LoginLogoutClient(automationContext).login();
+        userMgtClient = new UserManagementClient(backendUrl, sessionCookie);
+    }
+
+    @DataProvider(name = "scim2UserConfigProvider")
+    public static Object[][] scim2UserConfigProvider() {
+
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_ADMIN}, {TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        client = HttpClients.createDefault();
+    }
+
+    /*
+    Validate user listing with pagination.
+    */
+    @Test(dependsOnMethods = "testCreateUser")
+    public void testUserListingPagination() throws Exception {
+
+        HttpGet request = new HttpGet(getUserPath() + "?startIndex=" + START_INDEX + "&count=" + COUNT);
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 200, "Users " +
+                "has not been retrieved successfully");
```

Comment:
Ending dot of the sentense is required for the message.

Reply:
done